os/kernel/binary_manager/binary_manager_load.c

```diff
@@ -219,19 +220,22 @@ static int binary_manager_load(int bin_idx)
 #ifdef CONFIG_SUPPORT_COMMON_BINARY
 			if (bin_idx == BM_CMNLIB_IDX) {
 				ret = binary_manager_read_header(BINARY_COMMON, devpath, &common_header_data, true);
+				BIN_VER(bin_idx, BIN_USEIDX(bin_idx)) = common_header_data.version;
 			} else
 #endif
 			{
 				ret = binary_manager_read_header(BINARY_USERAPP, devpath, &user_header_data, true);
+				BIN_VER(bin_idx, BIN_USEIDX(bin_idx)) = user_header_data.version;
 			}
 			if (ret == BINMGR_OK) {
 				printf("[Binary Loading] %s Header Checking Success\n", BIN_NAME(bin_idx));
 			} else {
+				BIN_VER(bin_idx, BIN_USEIDX(bin_idx)) = 0;
```

Comment:
Could you leave the comment why binary version will be set to 0 for invalid case?